decidim-blogs/app/controllers/decidim/blogs/posts_controller.rb

```diff
@@ -13,20 +13,22 @@ class PostsController < Decidim::Blogs::ApplicationController
 
       def index; end
 
-      def show; end
+      def show
+        raise ActionController::RoutingError, "Not Found" unless post
```

Comment:
There is still one problem with this approach: as an admin user I might want to preview the blog post even before it is public.

So I think the single blog post page should be accessible for admins even when the post is not published yet.

The current functionality is correct for visitors and non-admins, i.e. show the 404 page.